### PR TITLE
🎄 V2 🎄

### PR DESCRIPTION
**Pull request by Diolor**

### CODE REVIEW

File: fotoapparat/src/main/java/io/fotoapparat/parameter/camera/CameraParameters.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        val previewResolution: Resolution
+) {
+    override fun toString(): String {
+        return "CameraParameters" + lineSeparator +
+                "flashMode:" + flashMode.wrap() +
+                "focusMode:" + focusMode.wrap() +
+                "jpegQuality:" + jpegQuality.wrap() +
+                "previewFpsRange:" + previewFpsRange.wrap() +
+                "antiBandingMode:" + antiBandingMode.wrap() +
+                "sensorSensitivity:" + sensorSensitivity?.wrap() +
```
Comment (jpribble) [inline fotoapparat/src/main/java/io/fotoapparat/parameter/camera/CameraParameters.kt]: FYI, a new line isn't added if the value is null so the display doesn't look quite right:

> Camera CameraParameters
> flashMode:		io.fotoapparat.parameter.Flash$Torch@743aeb
> focusMode:		io.fotoapparat.parameter.FocusMode$ContinuousFocusPicture@626481d
> jpegQuality:		90
> previewFpsRange:		FpsRange(min=15000, max=30000)
> antiBandingMode:		io.fotoapparat.parameter.AntiBandingMode$Auto@3644919
> **sensorSensitivity:nullpictureResolution:		Resolution(width=1280, height=720)**
> previewResolution:		Resolution(width=1280, height=720)

Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/parameter/camera/CameraParameters.kt]: Thanks, will fix it.

File: fotoapparat/src/main/java/io/fotoapparat/Fotoapparat.kt [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            )
+        }
+    }
+
+    /**
+     * Stops camera.
+     *
+     * @throws IllegalStateException If the camera has not started.
+     */
+    fun stop() {
```
Comment (jpribble) [inline fotoapparat/src/main/java/io/fotoapparat/Fotoapparat.kt]: The caller of stop() is not able to catch the IllegalStateException thrown if the camera has not been started. Can you update this to pass the exception back to the caller?

Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/Fotoapparat.kt]: This is by design. You must never call stop if you have not started the camera. Similar to an activity's lifecycle.

This lifecycle helps mental sanity. E.g. if you have a bug and call accidentally 2 times `stop()` or if e.g. you had forgotten to call `start()`, then if stop was optionally no-op and hiding exceptions you would scratch your head over those bugs. This way they are exposed.

Comment (jpribble) [reply fotoapparat/src/main/java/io/fotoapparat/Fotoapparat.kt]: Ok. By not propagating the IllegalStateException to caller, you're forcing users of the library to maintain their own flag of whether start() was called which feels less safe to me. I don't quite understand the logic of throwing an uncatchable fatal exception for a function that could just be a no-op if it's already stopped.

Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/Fotoapparat.kt]: Yes you are right. The docs & what I explained above is the desired logic. Due to the thread there it's not propagating the exception (incorrectly) making impossible to catch 👍

**Pull request merged**